### PR TITLE
Conformance test improvements

### PR DESCRIPTION
**Pull request by BenTheElder**
- don't dump logs in kind testing (not going to work..)
- `kind` [otherwise passes now], rename and move the CI job into conformance dashboards
- user a simpler (and faster!) build command in testgrid conformance docs

### CODE REVIEW

File: testgrid/conformance/README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ For running the conformance tests and obtaining the result files (`b)` and `c)`)
    ```sh
    git clone https://github.com/kubernetes/kubernetes.git && cd kubernetes && git checkout release-1.11
    ```
-   - run `make WHAT=test/e2e/e2e.test && make WHAT=ginkgo && make WHAT=cmd/kubectl` to build the test binaries
+   - run `make all WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"` to build the test binaries
```
Comment (BenTheElder) [inline testgrid/conformance/README.md]: you can see this in the `kind-e2e.sh` script. I've been meaning to make this change. this is faster as it doesn't involve repeatedly doing all of the docker build setup

File: config/jobs/kubernetes/sig-testing/kind.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          memory: "6Gi"
+    volumes:
+    - name: modules
+      hostPath:
+        path: /lib/modules
+        type: Directory
+    - name: cgroup
+      hostPath:
+        path: /sys/fs/cgroup
+        type: Directory
```
Comment (krzyzacy) [inline config/jobs/kubernetes/sig-testing/kind.yaml]: (new line)

**Comment by krzyzacy** [comment]
^^
/hold

File: config/jobs/kubernetes/sig-testing/kind.yaml (hunk 2 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-testing/kind.yaml]: done

**Pull request merged**